shared/hooks/useDateFilter.ts

```diff
@@ -0,0 +1,61 @@
+import { DateTime } from 'luxon'
+import { useState, useEffect } from 'react'
+
+import { EventModel } from 'shared/api'
+
+type UseDateFilterReturn = {
+  fromDate: DateTime | null
+  setFromDate: (fromDate: DateTime | null) => void
+  toDate: DateTime | null
+  setToDate: (toDate: DateTime | null) => void
+  filteredEvents: EventModel[]
+  fromDateError: string | null
+  toDateError: string | null
+}
+
+const useDateFilter = (events: EventModel[] | null, translation: (key: string) => string): UseDateFilterReturn => {
```

Comment:
Instead of passing the translation function, you could also just return the key and translate where it is used :)